src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double length;
+    private double width;
+
+    public Rectangle(String color, double length, double width) {
+        super(color);
+        this.length = length;
+        this.width = width;
+    }
+
+    @Override
+    public double getArea() {
+        return length * width;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, length: " + length + " units, width: " + width + " units, color: " + color);
```

Comment:
The `color` variable is protected in the `Figure` class, so it is not directly accessible in the `Rectangle` class. You need to access it through a getter method or directly from the superclass. Consider adding a `getColor()` method in the `Figure` class to retrieve the color.